src/test/ui/const-generics/auxiliary/impl-const.rs

```diff
@@ -0,0 +1,9 @@
+#![feature(const_generics)]
+
+pub struct Num<const N: usize>;
+
+// Braces around const expression causes crash
+impl Num<{5}> {
```

Comment:
The braces shouldn't matter, I wonder if this is a bug (cc @varkor).
What you want is to use an expression, like `2+3`, so that it definitely requires evaluation.

Reply:
We currently don't look into blocks for literals, It seems like we only do it for params at the moment

Reply:
Will get a PR ready which cleans this up.

Reply:
We shouldn't need braces here: arbitrary expressions should work, and `{5}` is an expression.

Reply:
The braces are needed to trigger the original ICE, so I included them in the test